test/com/amazon/ion/impl/IonReaderBinaryRawLargeStreamTest.java

```diff
@@ -70,4 +79,298 @@ public void testReadLargeScalarStream() throws Exception {
         }
         assertEquals(totalNumberOfBatches, batchesRead);
     }
+
+    @Test
+    public void readLargeScalarStreamNonIncremental() throws Exception {
+        readLargeScalarStream(IonReaderBuilder.standard());
+    }
+
+    @Test
+    public void readLargeScalarStreamIncremental() throws Exception {
+        readLargeScalarStream(IonReaderBuilder.standard().withIncrementalReadingEnabled(true));
+    }
+
+    @Test
+    public void readLargeContainer() throws Exception {
+        final Timestamp timestamp = Timestamp.forDay(2000, 1, 1);
+
+        byte[] data = testData(timestamp);
+
+        final int totalNumberOfBatches = (Integer.MAX_VALUE / data.length) + 32768;
+        ByteArrayOutputStream header = new ByteArrayOutputStream();
+        header.write(BINARY_VERSION_MARKER_1_0);
+        header.write(0xBE); // List with length subfield.
+        IonBinary.writeVarUInt(header, (long) data.length * totalNumberOfBatches); // Length
+        InputStream inputStream = new SequenceInputStream(
+            new ByteArrayInputStream(header.toByteArray()),
+            new RepeatInputStream(data, totalNumberOfBatches - 1) // This will provide the data 'totalNumberOfBatches' times
+        );
+
+        IonReader reader = IonReaderBuilder.standard().build(inputStream);
+        assertEquals(IonType.LIST, reader.next());
+        reader.stepIn();
+        int batchesRead = 0;
+        while (reader.next() != null) {
+            // Materializing on every iteration makes the tests take too long. Do it on every 100.
+            boolean materializeValues = (batchesRead % 100) == 0;
+            assertEquals(IonType.STRING, reader.getType());
+            if (materializeValues) {
+                assertEquals("foo", reader.stringValue());
+            }
+            assertEquals(IonType.DECIMAL, reader.next());
+            if (materializeValues) {
+                assertEquals(BigDecimal.TEN, reader.decimalValue());
+            }
+            assertEquals(IonType.TIMESTAMP, reader.next());
+            if (materializeValues) {
+                assertEquals(timestamp, reader.timestampValue());
+            }
+            batchesRead++;
+        }
+        assertNull(reader.next());
+        reader.stepOut();
+        assertNull(reader.next());
+        assertEquals(totalNumberOfBatches, batchesRead);
+    }
+
+    @Test
+    public void skipLargeContainer() throws Exception {
+        final Timestamp timestamp = Timestamp.forDay(2000, 1, 1);
+
+        byte[] data = testData(timestamp);
+
+        final int totalNumberOfBatches = (Integer.MAX_VALUE / data.length) + 17;
```

Comment:
What's the significance of `17`? Maybe add a comment?

Reply:
It's an arbitrary number that causes the size of the value to exceed Integer.MAX_VALUE. I can add a comment to that effect.

Reply:
Maybe better to use a value derived from `data.length`?